checkov/cloudformation/checks/resource/aws/RDSIAMAuthentication.py

```diff
@@ -0,0 +1,26 @@
+from checkov.cloudformation.parser.node import dict_node
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.cloudformation.checks.resource.base_resource_value_check import BaseResourceValueCheck
+
+
+class RDSIAMAuthentication(BaseResourceValueCheck):
+    def __init__(self) -> None:
+        name = "Ensure RDS database has IAM authentication enabled"
+        id = "CKV_AWS_161"
+        supported_resources = ["AWS::RDS::DBInstance"]
```

Comment:
if a cluster has it enabled, does the instance get it too automatically?

Reply:
yeah correct, in a cluster it is only possible to configure this on cluster level not instance level. Therefore I check for the engine type. Additionally it is only supported for mysql und postgresql.

Reply:
so we should only test an instance if the cluster has not configured it already? is that a graph query?

Reply:
no need, I added one check for the cluster and one for normal RDS Instanzen. In Terraform it is a bit more clear, what is what, because Terraform differentiate between them. In CloudFormation the main difference between a cluster instance and a normal instance is the cluster identifier and the engine type, if set.

Reply:
Hey @gruebel !

I'm hesitant to create this check for the DB instances - you can't create an instance without a cluster, and this should be on the cluster I believe. Am I wrong?

Reply:
Hey @nimrodkor 😄 
I think my explanation was not clear enough 😅
CloudFormation works a bit different compared to Terraform here. A single RDS instance and a cluster instance are provisioned with the same CloudFormation resource. When you want to deploy an instance to a cluster you have to set the cluster identifier property to reference the cluster. If you don't then it is a single RDS instance.
For cluster instances the engine type can only be something with aurora.